pkg/process/util/address.go

```diff
@@ -42,22 +63,20 @@ func AddressFromString(ip string) Address {
 // Warning: the returned `net.IP` will share the same underlying
 // memory as the given `buf` argument.
 func NetIPFromAddress(addr Address, buf []byte) net.IP {
-	if addrLen := addr.Len(); len(buf) < addrLen {
-		// if the function is misused we allocate
-		buf = make([]byte, addrLen)
-	}
-
 	n := addr.WriteTo(buf)
 	return net.IP(buf[:n])
 }
 
 // ToLowHigh converts an address into a pair of uint64 numbers
 func ToLowHigh(addr Address) (l, h uint64) {
-	if addr == nil {
-		return
-	}
+	bptr := IPBufferPool.Get().(*[]byte)
+	defer IPBufferPool.Put(bptr)
+
+	b := *bptr
+	n := addr.WriteTo(b)
```

Comment:
It would need to be verified, but I think if you use the `As4()` and `As16()` functions, which return arrays (not slices), then you can avoid the usage of the `sync.Pool` here, without allocation. I did as such here: https://github.com/DataDog/datadog-agent/pull/11870/files#diff-b05a1b719c4495a7656af892a26f4ea6c8dfef6a5131cc4c155032e8b49dd73aR73-R85